feat: update publish concept page (#248)
flox/floxdocs: docs/concepts/publishing.md

description: How to use Flox environments to build artifacts 
---

Once you've built an artifact with the [`flox build`][flox-build] command, you likely want to put it somewhere.
The [`flox publish`][flox-publish] command gives you the ability to upload artifacts to your private catalog.
In order to share packages with other people you must create an organization.

> See the [organizations][organizations-concept] page for more details.

Reviewer: nit wording, IMO the intend is to _use_, more specifically _install_ the package.
For that we need to make it available, i.e. put it into a private catalog.

Author: Fixed

- The artifact metadata
- The artifact itself

The artifact metadata is uploaded to Flox servers so that the Flox CLI can see that it's available via the [`flox search`][flox-search], [`flox show`][flox-show], and [`flox install`][flox-install] commands.
The artifact itself is uploaded to a Catalog Store.

A Catalog Store is effectively a cache for published artifacts, and Flox provides one by default.
An organization can choose to provide their own Catalog Store in the form of an S3-compatible storage provider.
In this case, it means that your organization has complete control over your artifacts and they will never be stored by Flox.
To pursue this option, contact Flox directly.

## Consuming published artifacts

Once you have uploaded an artifact via `flox publish`, the package becomes available in `flox search`, `flox show`, and `flox install`.
To distinguish these packages from those provided by the Base Catalog, published packages are prefixed with the name of the user or organization.
For example, if your user is called `myuser` and you publish an artifact named `hello`, the artifact will appear as `myuser/hello` in the Flox CLI.

> When a user runs `flox install myuser/hello`, the artifact is downloaded directly from the Catalog Store that it was published to, i.e. it is never downloaded to or cached on Flox servers.

Reviewer: ```suggestion
When a user runs `flox install myuser/hello`, the artifact is downloaded directly from the Catalog Store that it was published to. If organizations configure their own Catalog Store it is never downloaded to or cached on Flox servers.
```

Author: I'm finding it hard to walk the fine line between "say that BYOC exists
and document it" and "we want to strongly imply that publisher is the default workflow".

Author: Fixed
